src/main/java/fi/aalto/cs/apluscourses/intellij/utils/ListDependenciesPolicy.java

```diff
@@ -0,0 +1,26 @@
+package fi.aalto.cs.apluscourses.intellij.utils;
+
+import com.intellij.openapi.roots.LibraryOrderEntry;
+import com.intellij.openapi.roots.ModuleOrderEntry;
+import com.intellij.openapi.roots.RootPolicy;
+import java.util.List;
+import org.jetbrains.annotations.NotNull;
+
+public class ListDependenciesPolicy extends RootPolicy<List<String>> {
+
+  @Override
+  public List<String> visitModuleOrderEntry(@NotNull ModuleOrderEntry moduleOrderEntry,
+                                            List<String> value) {
+    value.add(moduleOrderEntry.getModuleName());
+    return value;
+  }
+
+  @Override
+  public List<String> visitLibraryOrderEntry(@NotNull LibraryOrderEntry libraryOrderEntry,
+                                             List<String> value) {
+    if (!libraryOrderEntry.isModuleLevel()) {
+      value.add(libraryOrderEntry.getLibraryName());
```

Comment:
will the `value`'s check for `null` make sense here (or `@NotNull`)?

Reply:
We cannot use `@NotNull` because the method is overridden.  We could add `if (value == null) return null` but I think that makes very little sense. Throwing an `ArgumentException` would be ok, but is that much more informative in stack trace than an `NullPointerException`? And then there would be a lot of redundant checks for null arguments.